### DISCUSSION TITLE
The total number of articles

### DISCUSSION THREAD

**Discussion by wxydejoy** [hexojs/hexo#4868] [Question / Answer]
I want to get the total number of articles on the deployed website. Is there any way?
At present, the known method is to find the number of links on the archive page

emmm
**i want to get other people's post number**

**Comment by stevenjoezhang** [comment]
You can pass the variable to front-end via template engine:
```njk
<script>
let total_number_of_articles = {{ site.posts.length }};
</script>
```

**Reply by SukkaW** [reply]
If you'd like to use the value internally, you can use `site.posts.length` in your theme directly.

If you want to expose the value in your front-end, use the method provided by @stevenjoezhang

**Reply by wxydejoy** [reply]
emmm
i want to get other people's post number

**Comment by wxydejoy** [comment]
emmm
i want to get other people's post number

**Comment by yoshinorin** [comment]
You can get it as follows, if you using [Hexo API](https://hexo.io/api/).

```js
const Hexo = require('hexo');
const hexo = new Hexo(process.cwd(), {silent: true});

hexo.init().then(() => {
  hexo.load().then(() => {

    const posts = hexo.locals.get('posts');
    console.log(posts.length);

    const pages = hexo.locals.get('pages');
    console.log(pages.length);
  }
}
```
